engine/flink/schemed-kafka-components-utils/src/test/scala/pl/touk/nussknacker/engine/schemedkafka/encode/BestEffortAvroEncoderSpec.scala

```diff
@@ -198,6 +199,18 @@ class BestEffortAvroEncoderSpec extends AnyFunSpec with Matchers with EitherValu
     BestEffortAvroEncoder(ValidationMode.lax).encodeRecord(Map("foo" -> "bar", "redundant" -> 15).asJava, schema) shouldBe 'valid
   }
 
+
+  it("should reject when missing required parameters") {
+    val schema = wrapWithRecordSchema(
+      """[
+        |  { "name": "foo", "type": "string" }
+        |]""".stripMargin)
+
+    BestEffortAvroEncoder(ValidationMode.strict).encodeRecord(Map("redundant" -> 15).asJava, schema) shouldBe 'invalid
+//    todo we should return invalid also in lax mode
```

Comment:
I wrote about that offline on slack. It is irrelevant TODO. This test is too much low level. You should check this using higher level, functional tests. If you do that, you'll see that everything works correctly. Encoder doesn't check it, but it is checked during validation so encoder never will require to check that. It is why I was asking to write functional tests. Encoding is only a technical detail. We should check user perspective.

Reply:
Check this debate: https://www.youtube.com/watch?v=0O_54WRKP1I :)

Reply:
yeah, I get it, I've removed this as it acs the same in lax and strict when filed is not provided